devops/jobs/DockerCI.groovy

```diff
@@ -0,0 +1,99 @@
+        extraVars.get("APPS_TO_REPO").each { app_name, app_repo ->
+            dslFactory.job('DockerCI/' + app_name) {
+
+                assert extraVars.containsKey('APPS_TO_REPO') : "Please define APPS_TO_REPO"
+                assert extraVars.containsKey('APPS_TO_BRANCH') : "Please define APPS_TO_BRANCH"
+                assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
+                assert extraVars.containsKey('CONFIGURATION_BRANCH') : "Please define CONFIGURATION_BRANCH"
+                assert extraVars.containsKey('EDX_REPO_ROOT') : "Please define EDX_REPO_ROOT"
+
+                // add credentials to log in to DockerHub; ID refers to credential ID as set up in Jenkins
+                wrappers {
+                    credentialsBinding {
+                        file("CONFIG_JSON_FILE", extraVars.get("CONFIG_JSON_FILE_CREDENTIAL_ID", "docker-config-json"))
+                    }
+                }
+                multiscm {
+                    // check out edx/configuration repository from GitHub; necessary to be able to run dependency analyzer
+                    git {
+                        remote {
+                            url(extraVars.get("CONFIGURATION_REPO_URL"))
+                            branch(extraVars.get("CONFIGURATION_BRANCH"))
+                        }
+                        extensions {
+                            relativeTargetDirectory('configuration')
+                            cleanAfterCheckout()
+                            // ignore notifications on commits to branch; this job will be triggered by
+                            // configuration-watcher job
+                            ignoreNotifyCommit()
+                            cloneOptions {
+                                // increase clone timeout to 3 hours
+                                timeout(360)
+                            }
+                        }
+                    }
+                    // if the IDA has a corresponding repository in the edx organization, checkout that repository
+                    if (app_repo) {
+                        git {
+                            remote {
+                                url(extraVars.get("EDX_REPO_ROOT") + app_repo + '.git')
+                                branch(extraVars.get("APPS_TO_BRANCH").get(app_name))
+                            }
+                            extensions {
+                                relativeTargetDirectory('edx')
+                                cloneOptions {
+                                    // increase clone timeout to 3 hours
+                                    timeout(360)
+                                }
+                            }
+                        }
+                    }
+                }
+            
+
+                // polls configuration repository for changes every 10 minutes
+                triggers {
+                    scm('H/10 * * * *')
+                }
```

Comment:
We're going to deploy this as polling, but since we have webhooks working on our jenkins (you can be triggered on merges to master) can we write a ticket to change to that later?